backend/cmd/api/main.go

```diff
@@ -94,13 +95,45 @@ func filteredLoaderFactory(db *sql.DB, queries *sqlc.Queries, collectionLoader *
 	}
 }
 
+var profilesLoaderCache = cache.New[uuid.UUID, *common.ProfileLoaders]()
+
+func getLoadersForProfile(queries *sqlc.Queries, profileID uuid.UUID) *common.ProfileLoaders {
+	if loaders, ok := profilesLoaderCache.Get(profileID); ok {
+		return loaders
+	}
+
+	profileQueries := queries.ProfileQueries(profileID)
+	loaders := &common.ProfileLoaders{
+		ProgressLoader: common.NewBatchLoader(profileQueries.GetProgressForEpisodes),
+	}
+
+	profilesLoaderCache.Set(profileID, loaders, cache.WithExpiration(time.Minute*5))
+
+	return loaders
+}
+
+func profileLoaderFactory(queries *sqlc.Queries) func(ctx context.Context) *common.ProfileLoaders {
+	return func(ctx context.Context) *common.ProfileLoaders {
+		ginCtx, err := utils.GinCtx(ctx)
+		if err != nil {
+			return nil
+		}
+		p := user.GetProfileFromCtx(ginCtx)
+		if p == nil {
+			return nil
+		}
+		return getLoadersForProfile(queries, p.ID)
+	}
+}
+
 // Defining the Graphql handler
 func graphqlHandler(db *sql.DB, queries *sqlc.Queries, loaders *common.BatchLoaders, searchService *search.Service, urlSigner *signing.Signer, config envConfig) gin.HandlerFunc {
 
 	resolver := graphapi.Resolver{
 		Queries:         queries,
 		Loaders:         loaders,
 		FilteredLoaders: filteredLoaderFactory(db, queries, loaders.CollectionLoader),
+		ProfileLoaders:  profileLoaderFactory(queries),
```

Comment:
Thoughts on this?

Very simple queries will be performed here, but will still be towards the database without batching across users.
Not sure how we would batch across profiles and not overcomplicate it.

Reply:
Let's do it like this for now. We can't batch things that don't make sense.